modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/StoriesViewModel.kt

```diff
@@ -49,25 +49,42 @@ class StoriesViewModel @Inject constructor(
 
     init {
         viewModelScope.launch {
-            storiesDataSource.loadStories().stateIn(viewModelScope).collect { result ->
-                cancelTimer()
-                if (result.size != stories.value.size) resetProgressAndCurrentIndex()
-                stories.value = result
-
-                val state = if (result.isEmpty()) {
-                    State.Error
-                } else {
-                    State.Loaded(
-                        currentStory = result[currentIndex],
-                        segmentsData = SegmentsData(
-                            xStartOffsets = List(numOfStories) { getXStartOffsetAtIndex(it) },
-                            widths = storyLengthsInMs.map { it / totalLengthInMs.toFloat() },
-                        )
-                    )
+            storiesDataSource.hasFullListeningHistory().stateIn(viewModelScope)
+                .collect { isFullListeningHistory ->
+                    if (!isFullListeningHistory) {
+                        storiesDataSource.syncListeningHistory().collect { success ->
```

Comment:
I'm not sure I understand this logic. The `hasFullListeningHistory()` stream will be suspended until `syncListeningHistory()` is finished collecting all items, and won't process anything from the upstream. Is this what should happen?

Reply:
I haven't hooked up `syncListeningHistory(...)` yet, and this might need some refactoring. The idea is to check if the user doesn't have a full listening history, get it from the server and then load all stories. 

You can leave it as is, it'll be taken care of while integrating the listening history endpoint.

Reply:
Would `syncListeningHistory()` be a continuous operation ore just a one shot one? If it is hitting an HTTP endpoint it most likely shouldn't be exposed as a `Flow` but rather as a suspending function. In any case it can be taken care of once the integration is in place.

Reply:
Correct, that should just be a suspending function.

Reply:
I added a TODO in 1ceb61f